examples/tv-casting-app/linux/main.cpp

```diff
@@ -328,21 +328,14 @@ class TargetVideoPlayerInfo
             .clientPool               = &gCASEClientPool,
         };
 
-        PeerId peerID           = fabric->GetPeerIdForNode(nodeId);
-        mOperationalDeviceProxy = chip::Platform::New<chip::OperationalDeviceProxy>(initParams, peerID);
+        PeerId peerID = fabric->GetPeerIdForNode(nodeId);
 
-        // TODO: figure out why this doesn't work so that we can remove OperationalDeviceProxy creation above,
-        // and remove the FindSecureSessionForNode and SetConnectedSession calls below
-        // mOperationalDeviceProxy = server->GetCASESessionManager()->FindExistingSession(nodeId);
+        mOperationalDeviceProxy = server->GetCASESessionManager()->FindExistingSession(peerID);
```

Comment:
Am in conversations with @chrisdecenzo to figure out a) what really failed here and b) if it is fixed now. Was not going to merge the PR till I got that confirmation.

Reply:
The code here is assuming that we can create an `OperationalDeviceProxy` instance and attach it immediately to a CASE session that just got established to us by the tv-app. While this will work most of the time, this is a dangerous assumption to make since it is entirely possible for that secure session to have been evicted in the time since that session was established to the point here when we desire to interact back with that peer. Up until this PR, `FindExistingSession` never actually re-used the underlying CASE session when creating an OperationalDeviceProxy instance, hence why this code did the kind of gymnastics it did to reach directly into the SessionManager, extract out the handle to the CASE session and manually create an instance of the proxy seeded with that handle.

I've fixed it now to use `FindOrEstablishSession`, but I cannot fix the code's assumption of synchronously having a valid session upon completion of that call. For now, this works because the callbacks will get synchronously invoked _most of the time_, but it's a fairly dangerous assumption to make.

I've filed a TODO there, @chrisdecenzo please prioritize fixing it.